docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,604 @@
+   names in *Names* by running [get-oob-ips] with *Names*
+   as input.
+
+6. Merge the set returned with *Name Servers*.
+
+7. Return *Name Servers*.
+
+### Outcome(s)
+
+Outcome is the set of name servers, where each unique name server name
+links to a possibly empty set of its IP addresses.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+-------------------------------------------------------------
+
+## Method: Get zone NS IP addresses
+
+### Method identifier
+**get-zone-ns-ips**
+
+### Objective
+
+Obtain the IP addresses of the name servers, as extracted from 
+the NS records of apex of the child zone.
+
+This is an external method to be used by test cases.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [get-zone-ns-names-and-ips].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Get the name servers set ("Name Servers") where each unique name 
+   server name is linked to a set of its possibly empty set of IP 
+   addresses by running [get-zone-ns-names-and-ips].
+
+2. Extract the IP addresses from *Name Servers* and create a
+   set of unique IP addresses.
+
+3. Return set of IP addresses.
+
+### Outcome(s)
+
+Outcome is a possibly empty set of IP addresses.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+-------------------------------------------------------------
+
+## Method: Get in-bailiwick address records in zone
+
+### Method identifier
+**get-ib-addr-in-zone**
+
+### Objective
+
+From the child zone, obtain the address records matching the
+[in-bailiwick] name server names found in the zone itself.
+
+This is an internal method to be used by other methods.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [get-del-ns-ips].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Using [get-del-ns-ips] get the IP addresses to the
+   name servers ("Name Server IPs").
+
+2. Using [get-zone-ns-names] get the names of the name
+   servers according to *Child Zone* itself 
+   ("Child Zone name server name").
+
+3. If none of the *Child Zone name server name* is an
+   [in-bailiwick] name server name, exit this method with an empty
+   set.
+
+4. Create a set name servers ("Name Servers") where each unique 
+   [in-bailiwick] name server name is linked to an empty set  
+   of IP addresses. 
+
+5. For each *Child Zone name server name* that is an [in-bailiwick]
+   name server:
+   1. Send an A and an AAAA query to all servers in *name server IPs*.
+   2. If a delegation (referral) to a sub-zone of Child Zone is returned, 
+      follow that delegation, possibly in several steps, by repeating the
+      A and AAAA queries.
+   3. Ignore non-referral responses unless AA flag is set. Cached data
+      is not accepted.
```

Comment:
Wouldn't it make more sense to move the instruction on cached data up to where the responses are collected?

Reply:
I'm referring to the wording "Cached data is not accepted". I believe it doesn't belongs here. I believe it belongs with the instruction to send out queries.